### ISSUE TITLE
Change decimation parameters from command line?

### ISSUE THREAD

**Issue by xymox12** [cdcseacave/openMVS#362]
v0.8 Windows 64

Hi

First - thanks for sharing this great application with us. 

I have often found that  by decimating the openMVS dense mesh in MeshLab using Quadric Edge Collapse decimation, but with altered settings to the defaults in openMVS, I get a more detailed, cleaner mesh than using the Refine step. It would be great if I could do this internally in openMVS rather than having to introduce meshlabserver into the workflow.

Is there a way to alter the settings from the command line or cfg file for the `vcg::tri::TriEdgeCollapseQuadricParameter` parameters seen from line 997 in /libs/MVS/Mesh.cpp? 

Thanks again.

**Comment by cdcseacave** [comment]
No, there is not a command line for that, though it would be easy to add. All parameters used in OpenMVS try to be general, good performance in all cases. So if you know decimation parameters that work in general better than those existent, you should directly modify the code. If you need to change the prams for each scene, in that case in my opinion there is an algorithmic problem that needs to be solved, and adding some more params will simply delay the problem. In other words I am not a big fan of adding new params, I add them only for a good reason (for instance the resolution at which the densification is computed sounds like a good param, trade speed for quality, and this is not necessary an algorithmic issue).

In conclusion, if you know better default params, pls change the code dirrectly, if there is a problem with the densification algo not being general enough, pls contribute by designing a better algorithm :)

**Comment by xymox12** [comment]
No probs - and thanks for the quick reply. I did what you suggested and hard coded the parameters - which was really easy thanks to your build setup, but the results were not what I expected (kind of fuzzy look because of the face angles). I think I may have to move the smooth function before the decimate in ReconstructMesh.exe to match the workflow of openmvs (no decimate, smooth=2) -> meshlab (decimate). 
Thanks again for a great app :)
